curvlinops/inverse.py

```diff
@@ -197,3 +204,120 @@ def _matvec(self, x: ndarray) -> ndarray:
                 )
 
         return self._scale * result
+
+
+class KFACInverseLinearOperator(_InverseLinearOperator):
+    """Class to invert instances of the ``KFACLinearOperator``."""
+
+    def __init__(
+        self,
+        A: KFACLinearOperator,
+        damping: Tuple[float, float] = (0.0, 0.0),
+        cache: bool = True,
+    ):
+        """Store the linear operator whose inverse should be represented.
+
+        Args:
+            A: ``KFACLinearOperator`` whose inverse is formed.
+            damping: Damping values for all input and gradient covariances.
+                Default: ``(0., 0.)``.
+            cache: Whether to cache the inverses of the Kronecker factors.
+                Default: ``True``.
+
+        Raises:
+            ValueError: If the linear operator is not a ``KFACLinearOperator``.
+        """
+        if not isinstance(A, KFACLinearOperator):
+            raise ValueError(
+                "The input `A` must be an instance of `KFACLinearOperator`."
+            )
+        super().__init__(A.dtype, A.shape)
+        self._A = A
+        self._damping = damping
+        self._cache = cache
+        self._inverse_input_covariances: Dict[str, Tensor] = {}
+        self._inverse_gradient_covariances: Dict[str, Tensor] = {}
+
+    def _compute_or_get_cached_inverse(self, name: str) -> Tuple[Tensor, Tensor]:
+        """Invert the Kronecker factors of the KFACLinearOperator or retrieve them.
+
+        Args:
+            name: Name of the layer for which to invert Kronecker factors.
+
+        Returns:
+            Tuple of inverses of the input and gradient covariance Kronecker factors.
+        """
+        if name in self._inverse_input_covariances:
+            aaT_inv = self._inverse_input_covariances.get(name)
+            ggT_inv = self._inverse_gradient_covariances.get(name)
+            return aaT_inv, ggT_inv
+        aaT = self._A._input_covariances.get(name)
+        ggT = self._A._gradient_covariances.get(name)
+        aaT_inv = (
+            cholesky_inverse(
+                cholesky(aaT + self._damping[0] * eye(aaT.shape[0], device=aaT.device))
+            )
+            if aaT is not None
+            else None
```

Comment:
Nit: Could invert statement to make it shorter: `None if aaT is None else ...`. Same below.